lib/Migration/Version1020Date20191002091034.php

```diff
@@ -0,0 +1,158 @@
+		]);
+		$messagesTable->addColumn('uid', 'integer', [
+			'notnull' => true,
+			'length' => 4,
+		]);
+		$messagesTable->addColumn('message_id', 'string', [
+			'notnull' => false,
+			'length' => 255,
+		]);
+		$messagesTable->addColumn('mailbox_id', 'string', [
+			'notnull' => true,
+			'length' => 4,
+		]);
+		$messagesTable->addColumn('subject', 'string', [
+			'notnull' => true,
+			'length' => 255,
+			'default' => '',
+		]);
+		$messagesTable->addColumn('sent_at', 'integer', [
+			'notnull' => true,
+			'length' => 4,
+		]);
+		$messagesTable->addColumn('flag_answered', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_deleted', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_draft', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_flagged', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_seen', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_forwarded', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_junk', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('flag_notjunk', 'boolean', [
+			'notnull' => true,
+			'default' => false,
+		]);
+		$messagesTable->addColumn('updated_at', 'integer', [
+			'notnull' => false,
+			'length' => 4,
+		]);
+		$messagesTable->setPrimaryKey(['id']);
+		// We allow each UID just once
+		$messagesTable->addUniqueIndex([
+			'uid',
+			'mailbox_id',
+		]);
+		$messagesTable->addIndex(['sent_at'], 'mail_message_sent_idx');
+
+		$recipientsTable = $schema->createTable('mail_recipients');
+		$recipientsTable->addColumn('id', 'integer', [
+			'autoincrement' => true,
+			'notnull' => true,
+			'length' => 20,
+		]);
+		$recipientsTable->addColumn('message_id', 'integer', [
+			'notnull' => true,
+			'length' => 20,
+		]);
+		$recipientsTable->addColumn('type', 'integer', [
+			'notnull' => true,
+			'length' => 2,
+		]);
+		$recipientsTable->addColumn('label', 'string', [
+			'notnull' => false,
+			'length' => 255,
+		]);
+		$recipientsTable->addColumn('email', 'string', [
+			'notnull' => true,
+			'length' => 255,
+		]);
+		$recipientsTable->setPrimaryKey(['id']);
+		$recipientsTable->addIndex(['message_id'], 'mail_recipient_msg_id_idx');
+		$recipientsTable->addIndex(['email'], 'mail_recipient_email_idx');
+
+		$mailboxTable = $schema->getTable('mail_mailboxes');
+		$mailboxTable->addColumn('sync_new_lock', 'integer', [
+			'notnull' => false,
+			'length' => 4,
+		]);
+		$mailboxTable->addColumn('sync_changed_lock', 'integer', [
+			'notnull' => false,
+			'length' => 4,
+		]);
+		$mailboxTable->addColumn('sync_vanished_lock', 'integer', [
+			'notnull' => false,
+			'length' => 4,
+		]);
+		$mailboxTable->dropColumn('sync_token');
```

Comment:
@nickvergessen I'm dropping `sync_token` here and create `sync_new_token`. For some reason the new column is where the old one was

```
+---------------------+--------------+------+-----+---------+----------------+
| Field               | Type         | Null | Key | Default | Extra          |
+---------------------+--------------+------+-----+---------+----------------+
| id                  | int(11)      | NO   | PRI | NULL    | auto_increment |
| name                | varchar(255) | NO   |     | NULL    |                |
| account_id          | int(11)      | NO   | MUL | NULL    |                |
| sync_new_token      | varchar(255) | YES  |     | NULL    |                |
| attributes          | varchar(255) | NO   |     | []      |                |
| delimiter           | varchar(1)   | NO   |     | NULL    |                |
| messages            | int(11)      | NO   |     | NULL    |                |
| unseen              | int(11)      | NO   |     | NULL    |                |
| selectable          | tinyint(1)   | NO   |     | 1       |                |
| special_use         | varchar(255) | NO   |     | []      |                |
| sync_new_lock       | int(11)      | YES  |     | NULL    |                |
| sync_changed_lock   | int(11)      | YES  |     | NULL    |                |
| sync_vanished_lock  | int(11)      | YES  |     | NULL    |                |
| sync_changed_token  | varchar(255) | YES  |     | NULL    |                |
| sync_vanished_token | varchar(255) | YES  |     | NULL    |                |
+---------------------+--------------+------+-----+---------+----------------+
```

Any possible explanation for that? I would expect to find the new col near the end of the list where the other new attributes are :thinking:

Reply:
(sync_new_token is also populated with values and I don't yet know where they are coming from. they are valid tokens. but I can't find the code path where we set only that one token)

Reply:
Well the order should not matter anyway?
But content being there already sounds strange....

Reply:
I don't care about the order. I'm just wondering as I have problems with the values.

Reply:
@ChristophWurst There's a good chance that Doctrine detects a rename here.

Reply:
You sure Doctrine does not simply detect a rename here?

Reply:
Yes, same conclusion @rullzer and I came to. Looks like it's trying to be smart and just renames the dropped column to one of the created once. Thus the data. Smells like https://github.com/doctrine/dbal/issues/3661 to me. The workaround is to split the migration into two, where the first one drops and the second creates. Seems to do the trick.

Reply:
That's quite a shocking change in their API. I guess we need to check all Migrations for this case.

Reply:
If you want to have a closer look you can use commit https://github.com/nextcloud/mail/commit/72b15d77e4d573faa6d6f4e0ede23f5d6f169c17. Just set up master and add an account. Then switch to that commit and run the migration.